cluster/cluster_config.go

```diff
@@ -17,112 +17,71 @@
 package cluster
 
 import (
-	"fmt"
 	"time"
 
-	"github.com/hazelcast/hazelcast-go-client/internal"
+	"github.com/hazelcast/hazelcast-go-client/types"
 )
 
-type Config struct {
-	loadBalancer         LoadBalancer
-	SecurityConfig       SecurityConfig
-	SSLConfig            SSLConfig
-	Name                 string
-	HazelcastCloudConfig HazelcastCloudConfig
-	Addrs                []string
-	InvocationTimeout    time.Duration
-	HeartbeatInterval    time.Duration
-	HeartbeatTimeout     time.Duration
-	ConnectionTimeout    time.Duration
-	DiscoveryConfig      DiscoveryConfig
-	RedoOperation        bool
-	SmartRouting         bool
-}
+const defaultName = "dev"
 
-func NewConfig() Config {
-	return Config{
-		Name:                 "dev",
-		Addrs:                []string{"127.0.0.1:5701"},
-		SmartRouting:         true,
-		ConnectionTimeout:    5 * time.Second,
-		HeartbeatInterval:    5 * time.Second,
-		HeartbeatTimeout:     60 * time.Second,
-		InvocationTimeout:    120 * time.Second,
-		SecurityConfig:       NewSecurityConfig(),
-		SSLConfig:            NewSSLConfig(),
-		HazelcastCloudConfig: NewHazelcastCloudConfig(),
-		DiscoveryConfig:      NewDiscoveryConfig(),
-	}
+type Config struct {
```

Comment:
I've tried to split the Go client into packages in a way that all code in those packages are related while also trying to keep cyclic dependency problems away. So, `cluster` package contains code that relates to the cluster, connections and connection bound invocations that are carried in those connections. I've tried to organize the configuration similarly, so `config.Cluster` (corresponds to `cluster.Config{}`) is configuration related to the cluster, etc. That allows me to pass an instance of `cluster.Config` to connection manager, cluster service, etc. `InvocationTimeout` is currently under `Cluster`, since that's where it is being used. I view the `Network` configuration which is something to be used by connections. So, IMO `HeartbeatXXX` and `Discovery` don't fit there. They seem more related to the cluster, e.g. `config.Cluster.Discovery.UsePublicIP` enables using public IPs of members in the cluster, etc. Similarly, `config.Cluster.HazelcastCloud` has options to enable retrieving members in a Hz Cloud cluster.

Reply:
OK, fair enough